### PR TITLE
feature(feature): filtering on admin roles

### PR DESCRIPTION
**Pull request by jhoward1994**
### What does it do?

Modify the route for retrieving all roles to allow filters

### Why is it needed?

Filtering roles from the FE settings page

### How to test it?

Provide information about the environment and the path to verify the behaviour.

### CODE REVIEW

File: packages/core/admin/server/services/role.js [hunk 1]
```diff
@@ -111,7 +111,8 @@ const find = (params = {}, populate) => {
  * @returns {Promise<array>}
  */
 const findAllWithUsersCount = async (populate) => {
-  const roles = await strapi.query('admin::role').findMany({ populate });
+  const roles = await strapi.entityService.findMany('admin::role', populate);
```
Comment (Marc-Roig) [inline packages/core/admin/server/services/role.js:114]: Why the change to use the entity service? Could it be possible to keep using the query engine?

Comment (jhoward1994) [reply packages/core/admin/server/services/role.js:114]: It's just that the sanitizedQuery produces a popualte object that is compatible with the entityService. It could work with the query engine though.

Would you do `const roles = await strapi.query('admin::role').findMany({ where: populate.filters });` ?

Comment (Marc-Roig) [reply packages/core/admin/server/services/role.js:114]: What format does it output? I think the query engine could take a populate argument too

Comment (jhoward1994) [reply packages/core/admin/server/services/role.js:114]: The populate object has this structure

```
{
  filters: {
    $and: [
      {
        name: {
          $contains: "value",
        },
      },
    ],
  },
}

```

Comment (Marc-Roig) [reply packages/core/admin/server/services/role.js:114]: Ah, I believe those are the filters of the query but that does not include a populate query. Are you sending a populate in your api request?

**Review by Marc-Roig — APPROVED** [review]
